ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java

```diff
@@ -542,6 +549,24 @@ private Long incrementalDump(Path dumpRoot, DumpMetaData dmd, Path cmRoot, Hive
       if (ev.getEventId() <= resumeFrom) {
         continue;
       }
+
+      //disable materialized-view replication if configured
+      String tblName = null;
+      tblName = ev.getTableName();
+      if(tblName != null) {
+        try {
+          Table table = null;
+          HiveWrapper.Tuple<Table> TableTuple = new HiveWrapper(hiveDb, dbName).table(tblName, conf);
+          table = TableTuple != null ? TableTuple.object : null;
+          if (TableTuple != null && TableType.MATERIALIZED_VIEW.equals(TableTuple.object.getTableType()) && !isMaterializedViewsReplEnabled()) {
```

Comment:
isMaterializedViewsReplEnabled check can be done at the top. no processing is needed if its enabled